File: Jongryul/01/11048.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = fs
  .readFileSync(filePath)
  .toString()
  .trim()
  .split("\n")
  .map((line) => line.replace("\r", ""));
const [n, m] = input.shift().split(" ").map(Number);
const arr = [Array.from({ length: m + 1 }, () => 0)];
const dp = [Array.from({ length: m + 1 }, () => 0)];
for (let i = 0; i < n; i++) {
  arr.push([0, ...input[i].split(" ").map(Number)]);
  dp.push(Array.from({ length: m + 1 }, () => 0));
}

for (let i = 1; i < n + 1; i++) {
  for (let j = 1; j < m + 1; j++) {
    dp[i][j] = Math.max(
      dp[i - 1][j] + arr[i][j],
      dp[i][j - 1] + arr[i][j],
      dp[i - 1][j - 1] + arr[i][j]
    );
  }
}

console.log(dp[n][m]);

File: Season 1/09th_Competition/Jongryul/01/11048.py
import sys
input = sys.stdin.readline
n, m = map(int, input().split())

arr = [list(map(int, input().split())) for _ in range(n)]
dp = [arr[0]]+[[0] * m for _ in range(n-1)]

for i in range(n):
	for j in range(m):
		for dr, dc in [(1, 0), (0, 1), (1, 1)]:
			r, c = dr + i, dc + j
			if r < n and c < m:
				dp[r][c] = max(dp[r][c], dp[i][j] + arr[r][c])


print(dp[n-1][m-1])